backend/routes/userAuth.js

```diff
@@ -0,0 +1,92 @@
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient();
+require('dotenv').config()
+
+const express = require('express');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const router = express.Router();
+
+const SECRET_KEY = process.env.SECRET_KEY
+
+//Middleware
+const authenticateToken = (req, res, next) => {
+  const token = req.cookies.token;
+
+  if (!token) {
+    return res.sendStatus(401);
+  }
+
+  jwt.verify(token, SECRET_KEY, (err, user) => {
+    if (err) {
+      return res.sendStatus(403);
+    }
+
+    req.user = user;
+    next();
+  });
+};
+
+// Definition of the routes for user registration, logining in and logging out
+// User Registration/Sign Up
+router.post('/users', async (req, res) => {
+    const {email, password, firstName, lastName} = req.body;
```

Comment:
should we use an email validator like this one? https://www.npmjs.com/package/email-validator

Reply:
I know you might be doing this in the frontend. But it is a good practice to catch errors in the frontend and backend.